ballerina/variable_validator.bal

```diff
@@ -242,21 +241,23 @@ class VariableValidator {
         argumentNode.setVariableDefinition(false);
     }
 
-    public isolated function setArgumentValue(json value, parser:ArgumentNode argument,
-                                              parser:VariableDefinitionNode varDef) {
-        if argument.getKind() == parser:T_INPUT_OBJECT {
-            argument.setVariableValue(value);
-        } else if argument.getKind() == parser:T_LIST {
-            argument.setVariableValue(value);
-        } else if argument.getKind() == parser:T_IDENTIFIER {
+    isolated function setArgumentValue(json value, parser:ArgumentNode argument, string variableTypeName,
+                                       __Type variableType) {
+        if getOfType(variableType).name.toString() == UPLOAD {
```

Comment:
Why do we need to use `toString()` here? Shall we check the usage of `toString()` and try to remove/reduce the usage of `toString()`?